cpp/arcticdb/storage/python_bindings.cpp

```diff
@@ -148,7 +149,7 @@ void register_bindings(py::module& storage, py::exception<arcticdb::ArcticExcept
         }, py::arg("library_path"), py::arg("throw_on_failure") = true)
         .def("remove_library_config", [](const LibraryManager& library_manager, std::string_view library_path){
             return library_manager.remove_library_config(LibraryPath{library_path, '.'});
-        })
+        }, py::call_guard<SingleThreadMutexHolder>())
```

Comment:
Why does this method need the mutex, and not the other methods in this class?

Reply:
I think I should have put the explanation in the description of this PR.
So basically, my understanding is, this will be short-lived patch. So my idea is keeping it safe for the short term:
1. No major rehaul
2. Add mutex to all functions in `PythonVersionStore`
3. If functions use folly::Future, which, in my eyes, can be converted async task easily. So 
`remove_key` is called by `remove_library_config`, which uses `folly::Future`, mutex is necessary